content/source/docs/plugin/framework/validation.html.md

```diff
@@ -0,0 +1,261 @@
+
+~> **NOTE:** When implementing validation logic, configuration values may be [unknown](./types.html#unknown) based on the source of the value. Implementations must account for this case, typically by returning early without returning new diagnostics.
+
+## Syntax and Basic Schema Validation
+
+The [Terraform configuration language](https://www.terraform.io/docs/language/) is declarative and an implementation of [HashiCorp Configuration Language](https://github.com/hashicorp/hcl) (HCL). The Terraform CLI is responsible for reading and parsing configurations for validity, based on Terraform's concepts such as `resource` blocks and associated syntax. Basic validation of value type and behavior information, for example returning an error when a string value is given where a list value is expected or returning an error when a required attribute is missing from a configuration, is automatically handled by the Terraform CLI based on the provider, resource, or data source schema.
+
+These syntax and basic schema checks are performed during the [`terraform apply`](https://www.terraform.io/docs/cli/commands/apply.html), [`terraform destroy`](https://www.terraform.io/docs/cli/commands/destroy.html), [`terraform plan`](https://www.terraform.io/docs/cli/commands/plan.html), and [`terraform validate`](https://www.terraform.io/docs/cli/commands/validate.html) commands. Any further validation provided by the framework occurs after these checks.
+
+## Attribute Validation
+
+It is common for provider implementations to introduce validation on attributes using the generic framework-defined types such as [`types.String`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types#String). The [`tfsdk.Attribute` type `Validators` field](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#Attribute.Validators) can be supplied with a list of validations and diagnostics will be returned from all validators. For example:
+
+```go
+// Typically within the tfsdk.Schema returned by GetSchema() for a provider,
+// resource, or data source.
+tfsdk.Attribute{
+    // ... other Attribute configuration ...
+
+    Validators: []AttributeValidators{
+        // These are example validators
+        stringLengthBetween(10, 256),
+        stringRegularExpression(regexp.MustCompile(`^[a-z0-9]+$`)),
+    },
+}
+```
+
+All validators will always be run, regardless of whether previous validators returned an error or not.
+
+-> The framework will implement or reference common use case attribute validations, such as string length, in the future.
```

Comment:
We probably don't want to say this only because it's making promises and we should avoid forward-looking statements. Instead, we probably just want to say that the framework does not currently reference common use case attribute validations. OR we omit this entirely :)

Reply:
Let's omit it for now. 😄